manager/controllers/app/m4dapplication_controller_test.go

```diff
@@ -469,28 +487,26 @@ var _ = Describe("M4DApplication Controller", func() {
 				_ = k8sClient.Get(context.Background(), appSignature, resource)
 				return resource.Status.Generated
 			}, timeout, interval).ShouldNot(BeNil())
+			By("Expecting plotter to be generated")
+			plotter := &apiv1alpha1.Plotter{}
+			Eventually(func() error {
+				key := types.NamespacedName{Name: resource.Status.Generated.Name, Namespace: resource.Status.Generated.Namespace}
+				return k8sClient.Get(context.Background(), key, plotter)
+			}, timeout, interval).Should(Succeed())
 
-			if resource.Status.Generated.Kind == "Blueprint" {
-				By("Expecting blueprint to be generated")
-				blueprint := &apiv1alpha1.Blueprint{}
-				Eventually(func() error {
-					Expect(k8sClient.Get(context.Background(), appSignature, resource)).Should(Succeed())
-					key := appSignature
-					key.Namespace = resource.Status.Generated.Namespace
-					return k8sClient.Get(context.Background(), key, blueprint)
-				}, timeout, interval).Should(Succeed())
-
-				// Check the generated blueprint
-				// There should be a single read module with two datasets
-				numReads := 0
-				for _, step := range blueprint.Spec.Flow.Steps {
-					if step.Template == readPathModule.Name {
-						numReads++
-						Expect(len(step.Arguments.Read)).To(Equal(2))
-					}
+			// Check the generated blueprint
```

Comment:
@shlomitk1 Why are these changes in this PR?

Reply:
I've fixed the tests along the way.